### DISCUSSION TITLE
How to get bleed for square images when using aspect ratio?

### DISCUSSION THREAD

**Discussion by keepitrio** [ValentinH/react-easy-crop#451] [Q&A]
When I set `aspect={1}`, if I use a non-square image, i can see the bleed (rest of the image being cropped out). But if I use a square image, that area is not available to me. See here: https://codesandbox.io/s/react-easy-crop-in-ts-forked-orh5so?file=/src/App.tsx You can switch line 26 to be from `image={SQUARE_IMAGE}` to `image={NON_SQUARE_IMAGE}`

I would like to have a container for the cropper which still shows more of the bleed image, but if I set a width and height to the container, it no longer shows the bleed. (uncomment line 15 and 16 in style sheet to see this)

Is there a solution for this?

**Comment by keepitrio** [comment]
I've read the threads about using `cropSize` not being recommended, but I'm wondering if there's a way to achieve something like this: The crop size always remains the same (for example, 400px by 400px), but no matter what image you use, the image is always flush to the edges of the cropper box. 

Right now if I set `cropSize={{ width: 400, height: 400}}` and upload a square or near square image, the entirety of the image does not show in the cropper. 
![Screenshot 2023-03-07 at 15 40 12](https://user-images.githubusercontent.com/14226804/223547265-e6bd84ff-c124-4dc4-a023-87a2e605ec5d.png)

**Reply by keepitrio** [reply]
and instead, if I set `aspectRatio={1}` and upload a square image, I get this: 
![Screenshot 2023-03-07 at 16 19 06](https://user-images.githubusercontent.com/14226804/223555241-9edcdf01-39ba-4c7a-b80e-dcf234af408e.png)

where i don't see the bleed anymore

**Reply by ValentinH** [reply]
I'm not sure I really get what you are looking for. 
if the aspect is set to be a square and the image is square, there is nothing to bleed as the cropper is perfectly adjusted to the image.

**Reply by keepitrio** [reply]
I forgot to include the detail that I also have zoom, rotation, flipping capability, along with showing the part of the image that's been cropped out.

So if I zoom into a square image, I'd like to see the rest of the image that's being cropped out. But if I use `aspect={1}`, the cropper box takes up the whole space and now I cannot see the part of the image that's being cropped out

**Reply by ValentinH** [reply]
OK I better understand. I'm playing with your example to see if I can get something that would fit what you are looking for.
BTW in your original message, you wrote "(uncomment line 15 and 16 in style sheet to see this)" but I don't see anything commented on those lines. Did you save your sandbox?

**Reply by keepitrio** [reply]
Thank you!! NOPE forgot to save, just added it back and saved!

**Reply by ValentinH** [reply]
Unfortunately, I couldn't find something fitting your needs.
I think we would need to add an option like `padding` that would add some spacing around the image inside the main container. However, I'm not really sure if this is something straightforward to do based on the way images are currently positioned in the cropper...

**Reply by keepitrio** [reply]
that makes sense. Thank you so much for looking into this, image handling is very complex!

**Reply by ValentinH** [reply]
Yes indeed 🙈
